### PR TITLE
refactor: improve CheckAvailableSpace reliability and output, fixes #8388

### LINKED ISSUE #8388: Detection of free space fails when df wraps the line
### Preliminary checklist

- [x] I am using the latest stable version of DDEV (see [upgrade guide](https://docs.ddev.com/en/stable/users/install/ddev-upgrade/))
- [x] I have searched [existing issues](https://github.com/ddev/ddev/issues)
- [x] I have checked the [troubleshooting guide](https://docs.ddev.com/en/stable/users/usage/troubleshooting/)
- [x] I have run `ddev utility test` or at least `ddev utility diagnose` to include output below

### Output of `ddev utility test` or `ddev utility diagnose`

<details><summary>Expand `ddev utility dockercheck` diagnostic information</summary>

```
Docker platform: linux-docker
docker buildx version 0.31.1 meets requirements for minimum 0.17.0
Using Docker context: default
Using Docker host: unix:///var/run/docker.sock
TLS not configured (no DOCKER_TLS_VERIFY or DOCKER_TLS)
docker-compose: v5.1.3
Docker version: 29.4.0
Docker API version: 1.54
Able to run simple container that mounts a volume.
Able to use internet inside container.
Your Docker install has only 0 available disk space, less than 5000000 warning level (0% used). Please increase disk image size. More info at https://docs.ddev.com/en/stable/users/usage/troubleshooting/#out-of-disk-space
Unable to perform trivial build with buildx: exit status 125; output=unknown flag: --quiet

Usage:  docker [OPTIONS] COMMAND [ARG...]

Run 'docker --help' for more information

Docker authentication is configured correctly
Docker provider checks completed with warnings. Please address the issues above.
```
</details>

On my system, the length of the device path exceeds what `df /` will display on a single line, leading the CheckAvailableSpace proc to use the incorrect filtering/fields (e.g. in my case, {print $3, $4;} would get the fields we want)

From Busybox v1.36.1 in the container (regardless of specifying columns, e.g. `docker run -it -e COLUMNS=$COLUMNS ddev/ddev-utilities:latest /bin/bash`) I get:
```
Filesystem           1K-blocks      Used Available Use% Mounted on
/dev/mapper/nvme0n1p2_crypt
                     1944559616 1625284868 231415612  88% /
```

I did note that `df -P /` appears to provide the desired output.

### Expected Behavior

231415612 should be seen as available K

### Actual Behavior

As per dockertest above, "Your Docker install has only 0 available disk space, less than 5000000 warning level (0% used). Please increase disk image size. More info at https://docs.ddev.com/en/stable/users/usage/troubleshooting/#out-of-disk-space" is reported.

### Steps To Reproduce

1. Given a device path exceeding ~23 characters
2. Run `ddev utility dockercheck`
3. See 0 space available

### Anything else?

Possible fix?

```diff
diff --git a/pkg/dockerutil/requirements.go b/pkg/dockerutil/requirements.go
index fddf29f2b..0ee141186 100644
--- a/pkg/dockerutil/requirements.go
+++ b/pkg/dockerutil/requirements.go
@@ -143,7 +143,7 @@ func CanRunWithoutDocker() bool {

 // CheckAvailableSpace outputs a warning if Docker space is low
 func CheckAvailableSpace() {
-       _, out, _ := RunSimpleContainer(versionconstants.UtilitiesImage, "check-available-space-"+util.RandString(6), []string{"sh", "-c", `df / | awk '!/Mounted/ {print $4, $5;}'`}, []string{}, []string{}, []string{}, "", true, false, nil, nil, nil)
+       _, out, _ := RunSimpleContainer(versionconstants.UtilitiesImage, "check-available-space-"+util.RandString(6), []string{"sh", "-c", `df -P / | awk '!/Mounted/ {print $4, $5;}'`}, []string{}, []string{}, []string{}, "", true, false, nil, nil, nil)
        out = strings.Trim(out, "% \r\n")
        parts := strings.Split(out, " ")
        if len(parts) != 2 {

```

### PR DESCRIPTION
**Pull request by wolcen**
## The Issue

- Fixes #8388

When a long device name was being tested for free space available, the output from df was wrapping. This caused the subsequent awk command not to see the free/total space, and following checks with the resulting variables to fail.

## How This PR Solves The Issue

- Replace `df / | awk` with `stat -f -c "%a %b %S"` run directly
  (no shell wrapper needed), avoiding awk dependency and df wrapping issues
- Return error from CheckAvailableSpace instead of calling util.Error directly;
  callers now display via util.Warning
- Switch MinimumDockerSpaceWarning from KB to bytes (SI: 5 * 1024 * 1024 * 1024)
- Show human-readable sizes in the space warning message

## Manual Testing Instructions

`ddev utility dockercheck` should no longer display the "Your Docker install has only 0 available disk space, less than 5000000 warning level (0% used). Please increase disk image size. More info at https://docs.ddev.com/en/stable/users/usage/troubleshooting/#out-of-disk-space" when run on a system with a device name exceeding ~23 characters.

## Automated Testing Overview

No test added. Not sure if this corner case deserves one/how one would do so...perhaps a loopback device with a long name? I don't know the exact length of the device name required to reproduce the issue.

## Release/Deployment Notes

N/A

### CODE REVIEW

**Comment by rfay** [comment]
Thanks for the PR. Can you suggest a way to reproduce the problem you're fixing (and reproduce the fix?)

**Comment by wolcen** [comment]
To be frank: no - I don't know how one would do that.

From what I understand:
- an assumption is made that / contains /var/lib/docker
- there is no reasonable method to "rename" the /dev that / exists on

A loop device obviously doesn't make sense, but a mapper device certainly could work e.g. https://linuxvox.com/blog/dev-mapper-linux/ - create a mapped device /dev/mapper/areallylongnamethatwillwrap. However, this would have to be mounted as /, which we certainly can't do, and also obviously doesn't make sense to do to a container.

I'm curious why we don't instead have a mount for e.g. `-v /var/lib/docker:/mnt` (or a resolved value, if different) and do a `df -P /mnt` to check the docker free space?

**Comment by rfay** [comment]
> I'm curious why we don't instead have a mount for e.g. -v /var/lib/docker:/mnt (or a resolved value, if different) and do a df -P /mnt to check the docker free space?

You're on Linux. The problem that this deals with is the traditional problem with Docker Desktop on macOS, where the docker disk space is fixed and not related to the actual available disk space. (On macOS and Windows, Docker runs inside a special Linux VM, and the code here has no way to access /var/lib/docker, and that's often not the correct filesystem. 

So what this check has done (and pretty successfully) is to get people to know when the overlay filesystem in `/` (as a proxy for the overall disk space available to Docker) is in trouble. It's been great for macOS. It even helps people on Linux sometimes when they're running out of total disk space. 

I'm glad you came up with such a nice and simple solution to this!

**Comment by rfay** [comment]
Please do test the artifacts in https://github.com/ddev/ddev/pull/8412#issuecomment-4446055370 on your system, thanks.

**Comment by wolcen** [comment]
Thanks for the explanation. The downloaded file is working for me on the system where I see the issue - no more warning:

```shell
$ ./ddev version
 ITEM              VALUE
 DDEV version      v1.25.2-41-g47de29629
 architecture      amd64
 cgo_enabled       0
 db                ddev/ddev-dbserver-mariadb-11.8:v1.25.2
 ddev-environment  linux
 ddev-ssh-agent    ddev/ddev-ssh-agent:v1.25.2
 docker            29.4.1
 docker-api        1.54
 docker-buildx     0.31.1
 docker-platform   linux-docker
 global-ddev-dir   /home/wolcen/.ddev
 go-version        go1.26.3
 mutagen           0.18.1
 os                linux
 router            ddev/ddev-traefik-router:v1.25.2
 web               ddev/ddev-webserver:20260508_rfay_faster_poweroff
 xhgui-image       ddev/ddev-xhgui:v1.25.2

Instrumentation is opted in, but AmplitudeAPIKey is not available. This usually means you have a locally-built ddev binary or one from a PR build. It's not an error. Please report it if you're using an official release build.
$ ./ddev utility dockercheck
Docker platform: linux-docker
docker buildx version 0.31.1 (/home/wolcen/.docker/cli-plugins/docker-buildx) meets requirements for minimum 0.17.0
Using Docker context: default
Using Docker host: unix:///var/run/docker.sock
TLS not configured (no DOCKER_TLS_VERIFY or DOCKER_TLS)
Docker version: 29.4.1
Docker API version: 1.54
Able to run simple container that mounts a volume.
Able to use internet inside container.
docker buildx is working correctly (trivial build succeeded)
Docker authentication is configured correctly
```

**Comment by stasadev** [comment]
I refactored the logic to rely only on stat instead of `df / | awk`:

```bash
stat -f -c "%a %b %S" /
```
gives raw block counts and block size directly.

And I made it human readable:

Before:
```
Your Docker install has only 3542896 available disk space, less than 5000000 warning level (35% used). Please increase disk image size.
```
After:
```
Your Docker install has only 3.5GB available disk space, less than 5.0GB warning level (35% used). Please increase disk image size.
```

**Comment by rfay** [comment]
Now it needs manual testing on a near-full or full Docker Desktop install, if you didn't already do that.

**Comment by stasadev** [comment]
Very easy to test without full disk, edit the `nodeps.MinimumDockerSpaceWarning` value and build the binary.

I have 154 GB free and set the limit to 155 GB:

```diff
-MinimumDockerSpaceWarning       = 5 * 1000 * 1000 * 1000
+MinimumDockerSpaceWarning       = 155 * 1000 * 1000 * 1000
```

```
$ ddev utility dockercheck
...
Warning: your Docker install has only 154.0GB available disk space, less than 155.0GB warning level (38% used). Please increase disk image size. More info at https://docs.ddev.com/en/stable/users/usage/troubleshooting/#out-of-disk-space
...
```

And lower the limit to 153 GB, should be no warning here:

```diff
-MinimumDockerSpaceWarning       = 5 * 1000 * 1000 * 1000
+MinimumDockerSpaceWarning       = 153 * 1000 * 1000 * 1000
```

```
$ ddev utility dockercheck
# No warning here
```

**Comment by stasadev** [comment]
Alternative testing: https://stackoverflow.com/a/257849/8097891

154 GB free is the same as 143 GiB (`fallocate` works with gibibytes, not gigabytes):

```
$ fallocate -l 140G /home/stas/bigfile

$ ddev utility dockercheck
...
Warning: your Docker install has only 3.6GB available disk space, less than 5.0GB warning level (98% used). Please increase disk image size. More info at https://docs.ddev.com/en/stable/users/usage/troubleshooting/#out-of-disk-space
...
```

And I created a second file, just to check the output in megabytes:

```
$ fallocate -l 3G /home/stas/bigfile2

$ ddev utility dockercheck
...
Warning: your Docker install has only 473.0MB available disk space, less than 5.0GB warning level (99% used). Please increase disk image size. More info at https://docs.ddev.com/en/stable/users/usage/troubleshooting/#out-of-disk-space
...
```

**Review by stasadev — APPROVED** [review]

**Comment by rfay** [comment]
On macOS/Docker Desktop (or Colima or Lima), where the host available space is not the issue, set the disk available to 48GB

image


```
ddev ssh
cd /var/www

rfay@d11-web:/var/www$ df -h .
Filesystem      Size  Used Avail Use% Mounted on
/dev/vda1        47G  5.6G   40G  13% /var/www
rfay@d11-web:/var/www$ fallocate -l 39G bigfile
rfay@d11-web:/var/www$ df -h .
Filesystem      Size  Used Avail Use% Mounted on
/dev/vda1        47G   44G  636M  99% /var/www
```

File: pkg/util/utils.go [hunk 1]
```diff
@@ -589,10 +589,13 @@ func ExtractCurlBody(curlOutput string) string {
...
 
-// FormatBytes converts bytes to a human-readable string
+// FormatBytes converts bytes to a human-readable string using SI decimal units (1000-based).
 // Returns format like: "2.3GB", "156.7MB", "1.5KB", "0B"
+// SI (metric):  1 KB  = 1000 bytes (KB/MB/GB)    - used by macOS Finder and storage manufacturers.
+// IEC 80000-13: 1 KiB = 1024 bytes (KiB/MiB/GiB) - used by Linux tools and Windows (mislabeled as GB).
+// We use SI because most DDEV users are on macOS, so numbers match what they see in Finder.
 func FormatBytes(bytes int64) string {
-	const unit = 1024
+	const unit = 1000
```
Comment (rfay) [inline pkg/util/utils.go]: This is a little questionable, should be thought about just a bit. Linux and Windows (for disk numbers) tend to use 1024 (and 1024*1024).  macOS (for disk numbers on host) tends to use 1000. 

I think this affects ParseDockerSystemDf(), and showAllMutagenVolumes() and CheckLargeFilesInSync(*) (in `ddev utility mutagen-diagnose`) 

I don't think this does much harm, but it also is a change very much unrelated to this bugfix. It's worth considering whether it's appropriate. It changes all the tests and the basic reporting, without relevance to the bugfix.

**Review by rfay — COMMENTED** [review]
Manual testing worked out fine. I guess I don't agree with the change from Linux/Windows filesystem sizing based on 1024 and MacOS-style filesystem numbers based on 1000. But if you think it's important, let's talk about it.

File: pkg/util/utils.go (hunk 1 above)
Comment (stasadev) [reply pkg/util/utils.go]: Okay, I'll switch back to 1024 in this PR.
